pkg/cluster/correct_cert_issuer.go

```diff
@@ -0,0 +1,68 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/util/keyvault"
+)
+
+// if the cluster isn't using a managed domain and has a DigiCert-issued
+// certificate, replace the certificate with one issued by OneCert. This
+// ensures that clusters upgrading to 4.16 aren't blocked due to the SHA-1
+// signing algorithm in use by DigiCert
+func (m *manager) correctCertificateIssuer(ctx context.Context) error {
+	if !strings.Contains(m.doc.OpenShiftCluster.Properties.ClusterProfile.Domain, ".") {
+		apiCertName := m.doc.ID + "-apiserver"
```

Comment:
Nit (and tangentially related to your changes): we could turn this into a "constant" by making a util method on the cluster manager type that returns the API cert's Secret name. This would ensure that it's correct across the different files that need to know and use the Secret name. Same applies to the ingress cert's Secret name.